File: ai-worker/app/embedding.py
"""Embedding 客户端：OpenAI 兼容 /v1/embeddings（SiliconFlow BGE-M3，1024 维）。"""
import httpx
from . import config


class EmbeddingError(Exception):
    pass


def embed_text(text: str) -> list[float]:
    """文本 → 1024 维向量。失败抛 EmbeddingError。"""
    try:
        r = httpx.post(
            f"{config.AI_BASE_URL}/embeddings",
            headers={
                "Authorization": f"Bearer {config.AI_API_KEY}",
                "Content-Type": "application/json",
            },
            json={"model": config.AI_EMBEDDING_MODEL, "input": text},
            timeout=30.0,
        )
    except httpx.HTTPError as e:
        raise EmbeddingError(f"network: {e}") from e
    if r.status_code != 200:
        raise EmbeddingError(f"embedding {r.status_code}: {r.text[:160]}")
    data = r.json()
    vec = (data.get("data") or [{}])[0].get("embedding")
    if not isinstance(vec, list) or len(vec) == 0:
        raise EmbeddingError("embedding empty")
    if len(vec) != config.AI_EMBEDDING_DIM:
        raise EmbeddingError(f"dim {len(vec)} != {config.AI_EMBEDDING_DIM}")
    return vec

File: src/lib/ai/embedding.ts
import { getEmbeddingConfig } from "./config";
import { AiProviderError } from "./provider";

/**
 * 调用 OpenAI 兼容 /v1/embeddings（默认 SiliconFlow BAAI/bge-m3，1024 维）。
 * 仅负责"文本→向量"。失败抛 AiProviderError，由 jobs.ts 捕获按重试策略处理。
 */
export async function embedText(text: string): Promise<{ vector: number[]; model: string }> {
  const cfg = getEmbeddingConfig();
  if (!cfg) throw new AiProviderError("embedding 未配置");

  let res: Response;
  try {
    res = await fetch(`${cfg.baseUrl}/embeddings`, {
      method: "POST",
      headers: {
        Authorization: `Bearer ${cfg.apiKey}`,
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ model: cfg.model, input: text }),
      signal: AbortSignal.timeout(30_000),
    });
  } catch (e) {
    throw new AiProviderError(`embedding 网络失败：${e instanceof Error ? e.message : String(e)}`);
  }

  if (!res.ok) {
    const detail = await res.text().catch(() => "");
    throw new AiProviderError(`embedding ${res.status}：${detail.slice(0, 160)}`);
  }

  const data = (await res.json()) as { data?: { embedding?: number[] }[] };
  const vec = data?.data?.[0]?.embedding;
  if (!Array.isArray(vec) || vec.length === 0) {
    throw new AiProviderError("embedding 返回为空");
  }
  if (vec.length !== cfg.dim) {
    throw new AiProviderError(`embedding 维度 ${vec.length} ≠ 配置 ${cfg.dim}`);
  }
  return { vector: vec, model: cfg.model };
}
